ReactNative/ReactTwitter/core/service/deep-linking-facade.ios.js

```diff
@@ -0,0 +1,30 @@
+import { Linking } from 'react-native'
+
+class DeepLinkingFacade {
+
+  static newInstance () {
+    return new DeepLinkingFacade(Linking)
+  }
+
+  constructor (linking) {
+    this.linking = linking
+  }
+
+  listenForDeepLinking () {
+    this.linking.removeEventListener('url', this._handleOpenURL)
+    this.linking.addEventListener('url', this._handleOpenURL)
+    return new Promise(function (resolve, reject) {
```

Comment:
Promises are for one shot, use-and-discard operations. What you're doing here is starting to listen to events the first time the button is clicked, then do something as soon as events come in.

_Ergo_, this should be an `Observable`, not a `Promise`.

But if this is only for demo reasons I would leave it like this.